code-challenges.test.js

```diff
@@ -51,17 +104,51 @@ const studyMinutesWeek2 = {
 
 // b) Create the function that makes the test pass.
 
+const objectValues = (obj) => {
+  return Object.values(obj).sort((a, b) => a - b)
+}   // output: 
+    //   PASS  ./jest.test.js
+    //   objectValues
+    //     ✓ returns an array of the values sorted from least to greatest") (2 ms)
+    //     ✓ "returns an array of the values sorted from least to greatest
+    
+    // Test Suites: 1 passed, 1 total
+    // Tests:       2 passed, 2 total
+    // Snapshots:   0 total
+    // Time:        0.150 s, estimated 1 s
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
+  // Pseudocode:
+    // Create function called accumulatedSum
+    // Create an empty array to hold the accumulated sums
+    // Use "for" loop to loop through each element in the input array
+    // Add the current elememnt's value to the current sum
+    // Use .push() to push the current sum to the result array
+    // Return the array of accumultd sums
+
 // a) Create a test with expect statements for each of the variables provided.
+describe('returns an array of the accumulating sum', () => {
+  expect(accumulatedSum([100, -17, -23, -9])).toEqual([100, 83, 60, 51]);
+  expect(accumulatedSum([250, -89, 100, -96])).toEqual([250, 161, 261, 165]);
+  expect(accumulatedSum([])).toEqual([]);
+})
 
-const accountTransactions1 = [100, -17, -23, -9]
+// const accountTransactions1 = [100, -17, -23, -9]
 // Expected output: [100, 83, 60, 51]
 
-const accountTransactions2 = [250, -89, 100, -96]
+// const accountTransactions2 = [250, -89, 100, -96]
 // Expected output: [250, 161, 261, 165]
 
-const accountTransactions3 = []
+// const accountTransactions3 = []
 // Expected output: []
 
 // b) Create the function that makes the test pass.
+
+const accumulatedSum = (array) => {
+  let sum = 0
+  return array.map = (number) => {
+    sum += number
+    return sum
+  }
+}
```

Comment:
Nice work on your coding challenges Nikki! One thing to note, if at first your pseudo code fails and you find a new solution make sure to update your pseudo to  reflect your new solution.

Reply:
Thank you! I always forget to change my pseudocode when I find a new way to work the problem. I'll definitely remember that!